tests/pytest_tests/unit_tests/test_lib/test_paths.py

```diff
@@ -0,0 +1,155 @@
+        ("et/tu.txt", "et/tu.txt", "et/tu.txt", b"et\\tu.txt"),
+        ("/ab/ra.txt", "/ab/ra.txt", "\\ab\\ra.txt", b"/ab/ra.txt"),
+        ("C:/a/b.txt", "C:/a/b.txt", "C:\\a\\b.txt", b"C:/a/b.txt"),
+    ],
+)
+def test_path_groups(target, posix, windows, _bytes):
+    assert target == LogicalPath(target)
+    assert target == LogicalPath(PurePosixPath(posix))
+    assert target == LogicalPath(PureWindowsPath(windows))
+    assert target == LogicalPath(_bytes) or platform.system() != "Windows"
+
+
+@given(fspaths())
+@example(b"")
+@example(".")
+@example("\\")
+@example(PureWindowsPath("C:/foo/bar.txt"))
+@example(PurePosixPath("x\\a/b.bin"))
+def test_path_conversion(path):
+    logical_path = LogicalPath(path)
+    assert isinstance(logical_path, str)
+    if platform.system() == "Windows":
+        assert "\\" not in logical_path
+
+    # For compatibility, enforce output identical to `to_forward_slash_path`.
+    if isinstance(path, str):
+        assert logical_path == to_forward_slash_path(path)
+
+
+@given(fspaths())
+def test_logical_path_is_idempotent(path):
+    norm_path = LogicalPath(path)
+    logical_path = LogicalPath(norm_path)
+    assert logical_path == LogicalPath(logical_path)
+    assert logical_path == LogicalPath(to_forward_slash_path(logical_path))
+    assert logical_path == to_forward_slash_path(LogicalPath(logical_path))
+
+
+@given(fspaths())
+def test_logical_path_round_trip(path):
+    logical_path = LogicalPath(path)
+    posix_path = logical_path.to_path()
+    assert isinstance(posix_path, PurePosixPath)
+    assert posix_path == LogicalPath(posix_path).to_path()
+    assert str(posix_path) == LogicalPath(posix_path) or not path
+
+
+@given(fspaths(), fspaths())
+def test_logical_path_acts_like_posix_path(path1, path2):
+    path1 = LogicalPath(path1)
+    assert path1.is_absolute() == PurePosixPath(path1).is_absolute()
+    assert path1.parts == PurePosixPath(path1).parts
+    assert not path1.is_reserved()
+    if path1.is_absolute():
+        assert path1.root == "/"
+        assert path1.as_uri() == PurePosixPath(path1).as_uri()
+        assert not path1.relative_to(path1.root).is_absolute()
+    else:
+        assert path1.anchor == ""
+
+    itself = path1.joinpath("bar").parent
+    assert isinstance(itself, LogicalPath)
+    assert PurePosixPath(itself) == PurePosixPath(path1)
+
+    path2 = LogicalPath(path2)
+    assert path1 / path2 == path1.joinpath(path2)
+    assert path1 / PurePosixPath("/foo") == LogicalPath("/foo")
+
+
+def test_sanitize_path_on_awful_input():
+    path = '\r/foo\u0000/AUX/<:?"*>/\0/\n\\\tCOM5.tar.gz /'
+    sanitized = sanitize_path(path)
+    assert sanitized == PurePosixPath("foo/_AUX/______/_COM5.tar.gz")
+
+
+@settings(max_examples=500)
+@given(one_of(fspaths(), text(), binary()))
+def test_sanitize_path_on_arbitrary_input(path):
+    try:
+        check_sanitized(sanitize_path(path))
+    except UnicodeDecodeError:
```

Comment:
How sure are you that we need this try/except? I've taken it out and run this test hundreds of times, and it hasn't failed once.

Reply:
Not? Maybe I needed it in one iteration of the code, but if it's not failing right now probably not.